jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java

```diff
@@ -906,4 +896,38 @@ else if (value.getClass().isArray())
         return sb.toString();
     }
 
+    /**
+     * Check that the offset and length of a subrange are within the size of a range.
+     * @param offset an offset of a subrange within a range of 0 to {code size} to start from.
+     *               Must be greater than or equal to 0 and less than the range size (if known).
+     * @param length the length of the subrange, -1 for the full length available from the offset,
+     *               otherwise must be greater than or equal to 0.
+     *               If the size is known, then the length will be reduced if necessary to fit within the size-offset.
+     * @param size the size of the range, or -1 if unknown.
+     * @return the length of the subrange, which may be calculated if length was less than 0 or if the length is limited by
+     *         a known size.
+     * @throws IndexOutOfBoundsException if the offset or length are out of range.
+     * @throws ArithmeticException if the length is not -1 and offset+length overflows.
+     */
+    public static long checkOffsetLengthSize(long offset, long length, long size)
+    {
+        if (length == 0)
+            return 0;
+
+        if (offset < 0)
+            throw new IndexOutOfBoundsException("Negative offset: " + offset);
+        if (length < -1)
```

Comment:
Could you please suffix your long constants with a `L` so that they look like `0L` or `-1L`? This makes it immediately obvious that we're dealing with `long`s and help reading the code.